### PR TITLE
Track stalls/loads separately in more providers

### PR DESCRIPTION
**Pull request by DonatoB**
This commit adds tracking to html5 youtube player, and all Flash providers (except RTMP)
[#90699216]

### CODE REVIEW

File: src/flash/com/longtailvideo/jwplayer/media/VideoMediaProvider.as [hunk 1]
```diff
@@ -290,8 +294,12 @@ public class VideoMediaProvider extends MediaProvider {
...
         // Toggle state between buffering and playing.
         if (_stream.bufferLength < 1 && state == PlayerState.PLAYING && _buffered < 100) {
-            setState(PlayerState.BUFFERING);
-        } else if (_stream.bufferLength > 1 && state == PlayerState.BUFFERING) {
+            if (this.seeking) {
+                setState(PlayerState.LOADING)
+            } else {
+                setState(PlayerState.STALLED);
+            }
+        } else if (_stream.bufferLength > 1 && (state == PlayerState.LOADING || state == PlayerState.STALLED)) {
```
Comment (robwalch) [inline src/flash/com/longtailvideo/jwplayer/media/VideoMediaProvider.as]: use ```>=``` and ```PlayerState.isBuffering(state)```

File: src/flash/com/longtailvideo/jwplayer/media/YouTubeMediaProvider.as [hunk 2]
```diff
@@ -125,7 +127,7 @@ public class YouTubeMediaProvider extends MediaProvider {
 
     /** Pause the YouTube movie. **/
     public override function pause():void {
-        if (state == PlayerState.PLAYING || state == PlayerState.BUFFERING) {
+        if (state == PlayerState.PLAYING || state == PlayerState.STALLED || state == PlayerState.LOADING) {
```
Comment (robwalch) [inline src/flash/com/longtailvideo/jwplayer/media/YouTubeMediaProvider.as]: PlayerState.isBuffering

File: src/flash/com/longtailvideo/jwplayer/media/YouTubeMediaProvider.as [hunk 3]
```diff
@@ -203,7 +207,7 @@ public class YouTubeMediaProvider extends MediaProvider {
         switch (Number(Object(evt).data)) {
             case 0:
                 // "ended"
-                if (state != PlayerState.BUFFERING && state != PlayerState.IDLE) {
+                if (state != PlayerState.LOADING && state != PlayerState.STALLED && state != PlayerState.IDLE) {
```
Comment (robwalch) [inline src/flash/com/longtailvideo/jwplayer/media/YouTubeMediaProvider.as]: PlayerState.isBuffering

File: src/js/providers/youtube.js [hunk 4]
```diff
@@ -50,8 +50,9 @@ define([
                 // always run this interval when not idle because we can't trust events from iFrame
                 _playingInterval = setInterval(_checkPlaybackHandler, 250);
                 if (state === states.PLAYING) {
+                    this.seeking = false;
                     _resetViewForMobile();
-                } else if (state === states.BUFFERING) {
+                } else if (state === states.LOADING || state === states.STALLED) {
```
Comment (robwalch) [inline src/js/providers/youtube.js]: PlayerState.isBuffering

File: src/flash/com/longtailvideo/jwplayer/player/PlayerState.as [hunk 5]
```diff
@@ -15,5 +15,15 @@ public class PlayerState {
...
     public static var PAUSED:String = "PAUSED";
+
+
+    // These are specific types of buffering which a provider may use for
+    //  better QOE introspection
+    public static const STALLED:String = "STALLED";
+    public static const LOADING:String = "LOADING";
+
+    public static function isBuffering(state:String):Boolean {
+        return (state === BUFFERING || state === STALLED || state === LOADING);
```
Comment (robwalch) [inline src/flash/com/longtailvideo/jwplayer/player/PlayerState.as]: check LOADING, STALLED, then BUFFERING

File: src/js/providers/youtube.js (hunk 4 above)
Comment (DonatoB) [inline src/js/providers/youtube.js]: This one is javascript

**Pull request merged**